### DISCUSSION TITLE
How to disable logging temporarily?

### DISCUSSION THREAD

**Discussion by jatentaki** [Lightning-AI/pytorch-lightning#10049] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
My `LightningModule.training_step` includes calls to `self.log` and finally returns the loss value. What is the best way to run `training_step` outside of a `Trainer` context, for debugging purposes (such as manual gradient inspection, etc)? Without the instrumentation by `Trainer`, logger is not defined and `self.log` calls cause an exception. I was trying to mock the logger to turn them to no-ops, but I wasn't successful.

**Comment by rohitgr7** [comment, marked as answer]
it was updated to warning recently: https://github.com/PyTorchLightning/pytorch-lightning/pull/9733
might work with recent release or master

**Reply by jatentaki** [reply]
Good to know, thanks!
